ledger-core/virtual/integration/method_test.go

```diff
@@ -1704,3 +1726,194 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
 
 	mc.Finish()
 }
+
+// b.Foo calls a.Bar, pulse changes, outgoing has proper token
+func TestVirtual_CallMethodOutgoing_WithPulseChange(t *testing.T) {
+	t.Log("C5141")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	logger := inslogger.FromContext(ctx)
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+		return execution.Request.CallSiteMethod
+	})
+	server.ReplaceRunner(runnerMock)
+
+	authService := authentication.NewServiceMock(mc)
+
+	authService.GetCallDelegationTokenMock.Set(func(outgoing reference.Global, to reference.Global, pn pulse.Number, object reference.Global) (c1 payload.CallDelegationToken) {
+		service := authentication.NewService(ctx, server.JetCoordinatorMock)
+		return service.GetCallDelegationToken(
+			outgoing, to, pn, object)
+	})
+	authService.HasToSendTokenMock.Return(true) // always send tokens
+	// skip token checks, in tests we have only one node, so we need to allow self signed tokens
+	authService.IsMessageFromVirtualLegitimateMock.Set(func(_ context.Context, payloadObj interface{}, sender reference.Global, pr pulse.Range) (mustReject bool, err error) {
+		return false, nil
+	})
+
+	server.ReplaceAuthenticationService(authService)
+
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	var (
+		barIsolation = contract.MethodIsolation{
+			Interference: contract.CallTolerable,
+			State:        contract.CallDirty,
+		}
+		fooIsolation = contract.MethodIsolation{
+			Interference: contract.CallTolerable,
+			State:        contract.CallDirty,
+		}
+
+		classA        = gen.UniqueGlobalRef()
+		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		callOutgoing = server.RandomLocalWithPulse()
+
+		outgoingCallRef = reference.NewRecordOf(objectBGlobal, callOutgoing)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+	Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectBGlobal).
+			CallMethod(objectAGlobal, classA, "Bar", []byte("123")).
+			SetInterference(barIsolation.Interference).
+			SetIsolation(barIsolation.State)
+		objectBExecutionMock := runnerMock.AddExecutionMock("Foo")
+		objectBExecutionMock.AddStart(
```

Comment:
I think that it is better to drop mock of executing B. Just don't resend VCallRequest, check token in the message and generate manually VCallResponse.

Reply:
not sure about this approach, It'll require more work, if something around VCallResponse will change.

Reply:
What if interface of execute adapter would change?

How this B's part is relevant to delegation token sent by call from A to B? As far as I know in this test the token is not checked by recipient. So it's irrelevant part. As I said many times messages are our boundaries for integration tests.

Reply:
switched to message instead of execution. Probably we need to rethink most of the tests in this file. Most of them actually execute more than one request.